src/com.docker.slirp/src/main.ml

```diff
@@ -44,72 +65,115 @@ let unix_listen path =
+      (fun fd ->
+        let flow = HV.connect fd in
+        Server.connect fs flow ()
+        >>= function
+        | Result.Error (`Msg m) ->
+          Log.err (fun f -> f "failed to establish 9P connection: %s" m);
+          Lwt.return ()
+        | Result.Ok server ->
+          Server.after_disconnect server
+      )
+  | _ ->
+    let module Ports = Active_list.Make(Forward_unix) in
+    let fs = Ports.make () in
+    Ports.set_context fs vsock_path;
+    let module Server = Protocol_9p.Server.Make(Log9P)(Host.Sockets.Stream.Unix)(Ports) in
+    unix_listen port_control_url
+    >>= fun port_s ->
+    Host.Sockets.Stream.Unix.listen port_s
+      (fun conn ->
+        Server.connect fs conn ()
+        >>= function
+        | Result.Error (`Msg m) ->
+          Log.err (fun f -> f "failed to establish 9P connection: %s" m);
+          Lwt.return ()
+        | Result.Ok server ->
+          Server.after_disconnect server
+    );
+    Lwt.return_unit
 
-let main_t socket_path port_control_path max_connections vsock_path db_path nofile pcap debug =
+let main_t socket_url port_control_url max_connections vsock_path db_path dns pcap debug =
   (* Write to stdout if expicitly requested [debug = true] or if the environment
      variable DEBUG is set *)
   let env_debug = try ignore @@ Unix.getenv "DEBUG"; true with Not_found -> false in
   if debug || env_debug then begin
     Logs.set_reporter (Logs_fmt.reporter ());
     Log.info (fun f -> f "Logging to stdout (stdout:%b DEBUG:%b)" debug env_debug);
   end else begin
-    let facility = Filename.basename Sys.executable_name in
-    let client = Asl.Client.create ~ident:"Docker" ~facility () in
-    Logs.set_reporter (Log_asl.reporter ~client ());
-    let dev_null = Unix.openfile "/dev/null" [ Unix.O_WRONLY ] 0 in
-    Unix.dup2 dev_null Unix.stdout;
-    Unix.dup2 dev_null Unix.stderr;
-    Log.info (fun f -> f "Logging to Apple System Log")
+    if Sys.os_type = "Win32" then begin
+      let h = Eventlog.register "Docker.exe" in
+      Logs.set_reporter (Log_eventlog.reporter ~eventlog:h ());
+      Log.info (fun f -> f "Logging to the Windows event log")
+    end else begin
+      let facility = Filename.basename Sys.executable_name in
+      let client = Asl.Client.create ~ident:"Docker" ~facility () in
+      Logs.set_reporter (Log_asl.reporter ~client ());
+      let dev_null = Unix.openfile "/dev/null" [ Unix.O_WRONLY ] 0 in
+      Unix.dup2 dev_null Unix.stdout;
+      Unix.dup2 dev_null Unix.stderr;
+      Log.info (fun f -> f "Logging to Apple System Log")
+    end
   end;
   Log.info (fun f -> f "Setting handler to ignore all SIGPIPE signals");
-  Sys.set_signal Sys.sigpipe Sys.Signal_ignore;
-  set_nofile nofile;
-  Log.info (fun f -> f "vpnkit version %%VERSION%% with hostnet version %s %s and uwt version %s"
-    Depends.hostnet_version Depends.hostnet_pinned Depends.uwt_version
+  (try Sys.set_signal Sys.sigpipe Sys.Signal_ignore with Invalid_argument _ -> ());
```

Comment:
Let me translate from Midwestern: there should be a comment here about why it's OK to ignore the exception.